metpy/gridding/tests/test_gridding_functions.py

```diff
@@ -0,0 +1,160 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import division
+
+import pytest
+
+from metpy.gridding.gridding_functions import (calc_kappa,
+                                               remove_observations_below_value,
+                                               remove_nan_observations,
+                                               remove_repeat_coordinates,
+                                               interpolate)
+
+import numpy as np
+
+from scipy.spatial.distance import cdist
+
+from numpy.testing import assert_array_almost_equal, assert_almost_equal
+
+from metpy.cbook import get_test_data
+
+
+@pytest.fixture()
+def test_coords():
+    x = np.array([8, 67, 79, 10, 52, 53, 98, 34, 15, 58], dtype=float)
+    y = np.array([24, 87, 48, 94, 98, 66, 14, 24, 60, 16], dtype=float)
+
+    return x, y
+
+
+def test_calc_kappa(test_coords):
+    r"""Tests calculate kappa parameter function"""
+
+    x, y = test_coords
+
+    spacing = np.mean((cdist(list(zip(x, y)),
+                             list(zip(x, y)))))
+
+    value = calc_kappa(spacing)
+
+    truth = 5762.6872048
+
+    assert_almost_equal(truth, value, decimal=6)
+
+
+def test_remove_observations_below_value(test_coords):
+    r"""Tests threshold observations function"""
+
+    x, y = test_coords[0], test_coords[1]
+
+    z = np.array(list(range(-10, 10, 2)))
+
+    x_, y_, z_ = remove_observations_below_value(x, y, z, val=0)
+
+    truthx = np.array([53, 98, 34, 15, 58])
+    truthy = np.array([66, 14, 24, 60, 16])
+    truthz = np.array([0, 2, 4, 6, 8])
+
+    assert_array_almost_equal(truthx, x_)
+    assert_array_almost_equal(truthy, y_)
+    assert_array_almost_equal(truthz, z_)
+
+
+def test_remove_nan_observations(test_coords):
+    r"""Tests remove observations equal to nan function"""
+
+    x, y = test_coords[0], test_coords[1]
+
+    z = np.array([np.nan, np.nan, np.nan, 1, 1, 1, 1, 1, 1, 1])
+
+    x_, y_, z_ = remove_nan_observations(x, y, z)
+
+    truthx = np.array([10, 52, 53, 98, 34, 15, 58])
+    truthy = np.array([94, 98, 66, 14, 24, 60, 16])
+    truthz = np.array([1, 1, 1, 1, 1, 1, 1])
+
+    assert_array_almost_equal(truthx, x_)
+    assert_array_almost_equal(truthy, y_)
+    assert_array_almost_equal(truthz, z_)
+
+
+def test_remove_repeat_coordinates(test_coords):
+    r"""Tests remove repeat coordinates function"""
+
+    x, y = test_coords
+
+    x[0] = 8.523
+    x[-1] = 8.523
+    y[0] = 24.123
+    y[-1] = 24.123
+
+    z = np.array(list(range(-10, 10, 2)))
+
+    x_, y_, z_ = remove_repeat_coordinates(x, y, z)
+
+    truthx = np.array([8.523, 67, 79, 10, 52, 53, 98, 34, 15])
+    truthy = np.array([24.123, 87, 48, 94, 98, 66, 14, 24, 60])
+    truthz = np.array([-10, -8, -6, -4, -2, 0, 2, 4, 6])
+
+    assert_array_almost_equal(truthx, x_)
+    assert_array_almost_equal(truthy, y_)
+    assert_array_almost_equal(truthz, z_)
+
+
+def test_interpolate(test_coords):
+    r"""Tests main interpolate function"""
```

Comment:
I really don't like the one gigantic test for all methods:

- Once the first assert fails, nothing else in the function runs
- There's a lot of very similar function calls

You can use pytest's parameterized decorator to do something better:
```python
interp_methods = ['natural_neighbor', 'cressman', 'barnes', 'linear', 'nearest', 'cubic', 'rbf']

@pytest.mark.parametrize('method', interp_methods)
def test_interpolate(test_coords, method):
    xp, yp = test_coords
    xp *= 10
    yp *= 10
    z = ....
    extra_kw = {}
    if method == 'cressman':
        extra_kw['search_radius'] = 200
    elif method == 'barnes':
        extra_kw['search_radius'] = 400

    _, _, img = interpolate(xp, yp, z, hres=10, interp_type=method, **extra_kw)

    truth = np.load(get_test_data('{0}_test.npz'.format(method)))['img']

    assert_array_almost_equal(truth, img)
```

This makes it much easier to update as any number of things change. It also converts to multiple tests, which is what this really is.